content/blog/2025-12-16-super-vectorization.md

```diff
@@ -0,0 +1,100 @@
++++
+title = "Predicated SSA and SLP vectorization"
+[extra]
+bio = """
+  Thomas is a undergraduate in the early M.Eng. program
+"""
+[[extra.authors]]
+name = "Thomas McFarland"
++++
+
+## Background
+
+### Classical Vectorization
+
+Vectorization is the classic example of SIMD parallelism. Without any confusing parallel primitives, programmers are able to achieve speedups. However, in classical vectorization programs must be structured in specific formats: control flow must be very simple, generally the expectation is that the operation is the only in its control block, and the operation must be isolated from other potentially confounding operations.
+
+Consider a simple loop that can be vectorized:
```

Comment:
Is there something you want readers to take away when looking at this example? It's a little weird to just say "consider this!" and then move on. Maybe you want to say that you'll be referring back to it later in the post, if that's true?